### DISCUSSION TITLE
about the command returned a non-zero exit code -9.

### DISCUSSION THREAD

**Discussion by whylovegithub** [apache/airflow#31294] [Q&A] [answered]
I am experiencing 
`[2023-05-15, 17:09:18 UTC] {subprocess.py:86} INFO - Output:
[2023-05-15, 17:19:41 UTC] {subprocess.py:97} INFO - Command exited with return code -9
[2023-05-15, 17:19:41 UTC] {taskinstance.py:1772} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/bash.py", line 196, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code -9.` 
in two of dags. This usually happens around 15 mins after I started the task. I tried to look in the source code of bash.py, And I just found the error code 1, 0 and 99. Try-Except I added in the code are also not triggered. So I wonder have anyone in the community have ever in countered this bug so that I can have a general direction to follow.

**Comment by potiuk** [comment]
It's the bash script of yours that executes wth -9. Likely it has a bug or something klills whatever your bash script is doing with SIGKILL - you should not look at Airflow but rather you need to understand what your bash script is doing and what's happening to it.

**Reply by whylovegithub** [reply]
I am basically using urlib3, request, json, pandas. So I dont really know where to find that -9.

**Comment by whylovegithub** [comment, marked as answer]
Our team finally identify this as a memory related problem. The code -9 actually generated by unix which means kill the process. Just incase someone encounter this and hope you can save some time.
